torch/csrc/jit/frontend/convert_to_ssa.cpp

```diff
@@ -201,6 +204,21 @@ struct EraseLoadStores {
           n->output()->replaceAllUsesWith(var);
           n->destroy();
         } break;
+        case prim::ComprehensionScope: {
```

Comment:
Since this isn't specific to comprehensions, I'd just name this `LocalVariableScope` or something similar to encourage its use anytime we need to introduce a temporary scope in the compiler. It's pretty funny that Python doesn't introduce variable scopes for for loops but does for for comprehensions.